library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -187,14 +188,47 @@ public boolean loadOwnedPurchasesFromGoogle() {
 	}
 
 	public boolean purchase(Activity activity, String productId) {
-		return purchase(activity, productId, Constants.PRODUCT_TYPE_MANAGED);
+		return purchase(activity, null, productId, Constants.PRODUCT_TYPE_MANAGED);
 	}
 
 	public boolean subscribe(Activity activity, String productId) {
-		return purchase(activity, productId, Constants.PRODUCT_TYPE_SUBSCRIPTION);
+		return purchase(activity, null, productId, Constants.PRODUCT_TYPE_SUBSCRIPTION);
 	}
 
-	private boolean purchase(Activity activity, String productId, String purchaseType) {
+	public boolean isSubscriptionUpdateSupported() {
+		try {
+			int response = billingService.isBillingSupported(Constants.GOOGLE_API_SUBSCRIPTION_CHANGE_VERSION, contextPackageName, Constants.PRODUCT_TYPE_SUBSCRIPTION);
+			return response == Constants.BILLING_RESPONSE_RESULT_OK;
+		} catch (RemoteException e) {
+			e.printStackTrace();
+		}
+		return false;
+	}
+
+	/**
+	 * Change subscription i.e. upgrade or downgrade
+	 *
+	 * @param oldProductId passing null or empty string will act the same as {@link #subscribe(Activity, String)}
+	 */
+	public boolean updateSubscription(Activity activity, String oldProductId, String productId) {
+		List<String> oldProductIds = null;
+		if (!TextUtils.isEmpty(oldProductId)) {
+			oldProductIds = new ArrayList<>(1);
+			oldProductIds.add(oldProductId);
+		}
+		return updateSubscription(activity, oldProductIds, productId);
```

Comment:
Just make a change to your branch and it will be automaticslly picked up by
the current PR

сб, 16 янв. 2016 г., 18:28, Mohd Zulfadhli Bin Ab Rashid <
notifications@github.com>:

> In library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java
> <https://github.com/anjlab/android-inapp-billing-v3/pull/142#discussion_r49932981>
> :
>
> > +		}
> > +		return false;
> > +	}
> > +
> > +	/**
> > +	 * Change subscription i.e. upgrade or downgrade
> > +	 *
> > +	 * @param oldProductId passing null or empty string will act the same as {@link #subscribe(Activity, String)}
> > +	 */
> > +	public boolean updateSubscription(Activity activity, String oldProductId, String productId) {
> > +		List<String> oldProductIds = null;
> > +		if (!TextUtils.isEmpty(oldProductId)) {
> > +			oldProductIds = new ArrayList<>(1);
> > +			oldProductIds.add(oldProductId);
> > +		}
> > +		return updateSubscription(activity, oldProductIds, productId);
>
> @serggl <https://github.com/serggl> This is my first time contributing to
> open source project. So, I make the changes and make another pull request?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anjlab/android-inapp-billing-v3/pull/142/files#r49932981>
> .
>
-- 
Sergey, AnjLab